test_scripts/AppServices/RPCPassThrough/002_SendLocation_Unsupported_Request.lua

```diff
@@ -69,26 +36,56 @@ local unsupportedResponse = {
   info = "Request CANNOT be handled by app services"
 }
 
-local coreResult = {success = true, resultCode = "SUCCESS", info = nil}
-
-local sendLocationRequest = { 
-  name = "SendLocation",
-  hmi_name = "Navigation.SendLocation",
+local rpcRequest = { 
   params = {
     longitudeDegrees = 50,
     latitudeDegrees = 50,
     locationName = "TestLocation" 
   },
-  hmi_params = params
+  hmi_params = {
+    longitudeDegrees = 50,
+    latitudeDegrees = 50,
+    locationName = "TestLocation" 
+  }
 }
 
-local sendLocationResponse = { 
-  name = "SendLocation",
-  hmi_name = "Navigation.SendLocation",
-  params = coreResult,
-  hmi_params = { code = 0 }
+local rpcResponse = { 
+  params = {success = true, resultCode = "SUCCESS", info = nil},
+  hmi_params = {code = 0}
 }
 
+--[[ Local functions ]]
+local function PTUfunc(tbl)
+  --Add permissions for app1
+  local pt_entry = common.getAppServiceProducerConfig(1)
+  pt_entry.app_services.NAVIGATION = { handled_rpcs = {{function_id = 39}} }
+  tbl.policy_table.app_policies[common.getConfigAppParams(1).fullAppID] = pt_entry
+  --Add permissions for app2
+  pt_entry = common.getAppDataForPTU(2)
+  pt_entry.groups = { "Base-4" , "SendLocation" }
+  tbl.policy_table.app_policies[common.getConfigAppParams(2).fullAppID] = pt_entry
+end
+
+local function RPCPassThruTest()
+  local providerMobileSession = common.getMobileSession(1)
+  local mobileSession = common.getMobileSession(2)
+
+  local cid = mobileSession:SendRPC("SendLocation", rpcRequest.params)
+      
+  providerMobileSession:ExpectRequest("SendLocation", rpcRequest.params):Do(function(_, data)
+    providerMobileSession:SendResponse("SendLocation", data.rpcCorrelationId, unsupportedResponse)
+  end)
+
+  --Core will handle the RPC
+  EXPECT_HMICALL("Navigation.SendLocation", rpcRequest.hmi_params):Times(1)
+  :Do(function(_, data)
+    common.getHMIConnection():SendResponse(data.id, data.method, rpcResponse.hmi_params.code, rpcResponse.hmi_params)
```

Comment:
Including the final argument is redundant here, since `hmi_params` only contains code, and that is set by the third argument.

Reply:
In any case, `code` should probably be separated from params, since it is a separate param in this function